Mathlib/LinearAlgebra/AffineSpace/ContinuousAffineEquiv.lean

```diff
@@ -0,0 +1,295 @@
+/-
+Copyright (c) 2024 Michael Rothgang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Rothgang
+-/
+import Mathlib.LinearAlgebra.AffineSpace.AffineEquiv
+import Mathlib.Topology.Algebra.Module.Basic
+
+/-!
+# Continuous affine equivalences
+
+In this file, we define continuous affine equivalences, affine equivalences
+which are continuous with continuous inverse.
+
+## Main definitions
+* `ContinuousAffineEquiv.refl k P`: the identity map as a `ContinuousAffineEquiv`;
+* `e.symm`: the inverse map of a `ContinuousAffineEquiv` as a `ContinuousAffineEquiv`;
+* `e.trans e'`: composition of two `ContinuousAffineEquiv`s; note that the order
+  follows `mathlib`'s `CategoryTheory` convention (apply `e`, then `e'`),
+  not the convention used in function composition and compositions of bundled morphisms.
+
+* `e.toHomeomorph`: the continuous affine equivalence `e` as a homeomorphism
+* `ContinuousLinearEquiv.toContinuousAffineEquiv`: a continuous linear equivalence as a continuous
+  affine equivalence
+* `ContinuousAffineEquiv.constVAdd`: `AffineEquiv.constVAdd` as a continuous affine equivalence
+
+## TODO
+- equip `ContinuousAffineEquiv k P P` with a `Group` structure,
+with multiplication corresponding to composition in `AffineEquiv.group`.
+
+-/
+
+open Function
+
+/-- A continuous affine equivalence between two affine topological spaces is an affine equivalence
+such that forward and inverse maps are continuous. -/
+structure ContinuousAffineEquiv (k P₁ P₂ : Type*) {V₁ V₂ : Type*} [Ring k]
+  [AddCommGroup V₁] [Module k V₁] [AddTorsor V₁ P₁] [TopologicalSpace P₁]
+  [AddCommGroup V₂] [Module k V₂] [AddTorsor V₂ P₂] [TopologicalSpace P₂] extends P₁ ≃ᵃ[k] P₂ where
+  continuous_toFun : Continuous toFun := by continuity
+  continuous_invFun : Continuous invFun := by continuity
```

Comment:
Should these be:
```suggestion
  continuous_toFun : Continuous toFun := by fun_prop
  continuous_invFun : Continuous invFun := by fun_prop
```
(ask on Zulip if you're not sure either)

Reply:
Good question: as-is, this failed to prove continuity for `refl`, so I'm leaving it as-is. (But asking why would be a great follow-up question!)